File: dream/sleep.py
# Beowulf - sleep.py
# Move data to and from dream (Python <-> Nodejs)

# Not every part of the cluster can or will dream...

#...we dream in python so we can use tensorflow and scikit...

#... parts of the cluster that cannot execute python will not dream.

#... Needless to say the dream is not ready yet. Still forging the algorithms. Stay tuned.

import sys
import json
import numpy as np

# Read data from stdin
def read_in():
    lines = sys.stdin.readlines()
    # parse data
    return json.loads(lines[0])

def main():
    # get data as an array from read_in()
    lines = read_in()

    np_lines = np.array(lines)

    # use numpys sum method to test return 
    lines_sum = np.sum(np_lines)

    # return the sum to the output stream
    print lines_sum

# start process
if __name__ == '__main__':
    main()

File: sleep.js
// Beowulf - sleep.js
// Shut down sensations. Move data to and from the dream. (Nodejs <-> Python)

// Not every part of the cluster can or will dream...

//...we dream in python so we can use tensorflow and scikit...

//... parts of the cluster that cannot execute python will not dream.

//... Needless to say the dream is not ready yet. Still forging the algorithms. Stay tuned.

var senses = require('./senses.js')

senses.exit()

var spawn = require('child_process').spawn
var py = spawn('python', ['dream/sleep.py'])
var higher = require('./high.js')

var data = higher.data
var dataString = ''

py.stdout.on('data', function(data){
  dataString += data.toString()
});
py.stdout.on('end', function(){
  console.log('Sum of numbers=',dataString)
});
py.stdin.write(JSON.stringify(data))
py.stdin.end()